File: test-suite-python/src/test/python/micronaut/docs/metrics/configurer/SimpleMeterRegistryConfigurer.py
from micronaut.context.annotation import Requires

# tag::imports[]
from io.micrometer.core.instrument.simple import SimpleMeterRegistry
from jakarta.inject import Singleton
from micronaut.configuration.metrics.aggregator import MeterRegistryConfigurer
from micronaut.configuration.metrics.annotation import RequiresMetrics
from micronaut.core.annotation import Order
# end::imports[]


@Requires(property="spec.name", value="SimpleMeterRegistryConfigurerTest")
# tag::class[]
@Order(2147483647)
@Singleton
@RequiresMetrics
class SimpleMeterRegistryConfigurer(MeterRegistryConfigurer[SimpleMeterRegistry]):

    def configure(self, meterRegistry: SimpleMeterRegistry) -> None:
        meterRegistry.config().commonTags("key", "value")

    def getType(self) -> type[SimpleMeterRegistry]:
        return SimpleMeterRegistry
# end::class[]

File: test-suite/src/test/java/io/micronaut/docs/metrics/configurer/SimpleMeterRegistryConfigurer.java
package io.micronaut.docs.metrics.configurer;

// tag::imports[]
import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
import io.micronaut.configuration.metrics.aggregator.MeterRegistryConfigurer;
import io.micronaut.configuration.metrics.annotation.RequiresMetrics;
import io.micronaut.core.annotation.Order;
import jakarta.inject.Singleton;
// end::imports[]
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "SimpleMeterRegistryConfigurerTest")
// tag::class[]
@Order(Integer.MAX_VALUE)
@Singleton
@RequiresMetrics
public class SimpleMeterRegistryConfigurer implements MeterRegistryConfigurer<SimpleMeterRegistry> {

    @Override
    public void configure(SimpleMeterRegistry meterRegistry) {
        meterRegistry.config().commonTags("key", "value");
    }

    @Override
    public Class<SimpleMeterRegistry> getType() {
        return SimpleMeterRegistry.class;
    }
}
// end::class[]
